File: 165-compare-version-numbers/165-compare-version-numbers.java
class Solution {
    public int compareVersion(String v1, String v2) {
        String[] l1 = v1.split("\\.");
        String[] l2 = v2.split("\\.");
        
        int len = Math.max(v1.length(), v2.length());
        for (int i = 0; i < len; i++) {
            Integer c1 = (i < l1.length) ? Integer.parseInt(l1[i]) : 0;
            Integer c2 = (i < l2.length) ? Integer.parseInt(l2[i]) : 0;
            
            if (c1 > c2)
                return 1;
            if (c2 > c1)
                return -1;
        }
        return 0;
    }
}

File: 165-compare-version-numbers/165-compare-version-numbers.cpp
class Solution {
public:
    int compareVersion(string v1, string v2) {
        for (auto& v : v1)
            if (v == '.')
                v = ' ';
        for (auto& v : v2)
            if (v == '.')
                v = ' ';
        
        istringstream s1(v1), s2(v2);
        
        while (1) {
            int n, m;
            if (!(s1 >> n))
                n = 0;
            if (!(s2 >> m))
                m = 0;
            if (!s1 && !s2)
                return 0;
            if (n > m)
                return 1;
            if (m > n)
                return -1;
        }
    }
};